### ISSUE TITLE
[W::bam_hdr_read] EOF marker is absent. The input is probably truncated.

### ISSUE THREAD

**Issue by apfuentes** [ekg/freebayes#382]
Hi Erik,
I am trying to run freebayes to call SNPs in 17 pool-seq files (41-50 individuals per pool, varying ploidy from 82 to 100, diploid organism). This is the command I used:
```
freebayes -f /PATH/Ref_genome/ref.fasta \
--ploidy 100 \
--use-best-n-alleles 4 \
--pooled-discrete \
--min-alternate-fraction 0.06 \
-b /PATH/1.sorted.MarkDup.RG.bam \
-b /PATH/2.sorted.MarkDup.RG.bam \
-b /PATH/3.sorted.MarkDup.RG.bam \
-b /PATH/4.sorted.MarkDup.RG.bam \
-b /PATH/5.sorted.MarkDup.RG.bam \
-b /PATH/6.sorted.MarkDup.RG.bam \
-b /PATH/7.sorted.MarkDup.RG.bam \
-b /PATH/8.sorted.MarkDup.RG.bam \
-b /PATH/9.sorted.MarkDup.RG.bam \
-b /PATH/10.sorted.MarkDup.RG.bam \
-b /PATH/11.sorted.MarkDup.RG.bam \
-b /PATH/12.sorted.MarkDup.RG.bam \
-b /PATH/13.sorted.MarkDup.RG.bam \
-b /PATH/14.sorted.MarkDup.RG.bam \
-b /PATH/15.sorted.MarkDup.RG.bam \
-b /PATH/16.sorted.MarkDup.RG.bam \
-b /PATH/17.sorted.MarkDup.RG.bam \
-v /PATH/Freebayes_res/WG.17.pools.raw.vcf
```

But I am getting this message (and the program stops there):
```
[W::bam_hdr_read] EOF marker is absent. The input is probably truncated.
```

Any idea what is causing this? Each BAM file was mapped with BWA-MEM, then sorted, marked duplicates, and read groups added using Picard tools. This is an example of the command used to add read groups:

java -Xmx50g -jar ~/picard-tools-2.5.0/picard.jar \
AddOrReplaceReadGroups \
INPUT=BDO.sorted.MarkDup.bam \
OUTPUT=BDO.PoolSeq.sorted.MarkDup.RG.bam \
VALIDATION_STRINGENCY=SILENT \
RGID=BDO RGLB=BDO RGPL=BDO RGPU=BDO RGSM=BDO

Thanks for any help!

Angela

**Comment by krepsen** [comment]
Hi Angela,
Have you figured out this problem? 
I encounter the same issue, but only when I try to run larger data sets, with only a few bam files the Freebayes runs without problem.
Please let me know if you found out what causes this error.

Cheers,
Kristian

**Comment by apfuentes** [comment]
Hi Kristian,

It seems this message indicates the BAM files are missing a 28 bytes code at the end (either a known bug for ungizziped BAM files in an old version of Samtools, or the BAM files are truncated...not sure which one could apply to my case). Despite the message, the program seems to work as I noticed an spike in RAM memory usage (I don't know if the job was successful or not as I run out of RAM memory and had to kill it before ending).

Hope that helps.

Cheers,
Angela

**Comment by krepsen** [comment]
Hi Angela,

Thanks, I came to the same conclusion, something about an error in an earlier version of Samtools.
I found this [http://seqanswers.com/forums/showthread.php?t=15363](url) and checked a few of my BAM files to confirm correct ending -
 making sure they were indeed not truncated, but with that weird EOF marker/bam header problem. Now I just have to be patient to see if the freebayes (and my computer) manage to complete the SNP calling.
Thanks again.

Kristian
